packages/desktop-client/src/components/transactions/TransactionsTable.tsx

```diff
@@ -2069,589 +2227,691 @@ function TransactionTableInner({
+
+    const { onAddSplit: onAddSplitProp } = props;
+
+    const onAddSplit = useCallback(
+      (id: TransactionEntity['id']) => {
+        const {
+          tableNavigator,
+          newNavigator,
+          newTransactions: newTrans,
+        } = latestState.current;
+
+        if (isTemporaryId(id)) {
+          const { data, diff } = addSplitTransaction(newTrans, id);
+          setNewTransactions(data);
           newNavigator.onEdit(
             diff.added[0].id,
             latestState.current.newNavigator.focusedField,
           );
-        }
-      } else {
-        const trans = latestState.current.transactions.find(t => t.id === id);
-        const newId = onSplitProp(id);
-
-        splitsExpandedDispatch({ type: 'open-split', id: trans.id });
-
-        const { tableNavigator } = latestState.current;
-        if (trans.amount === null) {
-          tableNavigator.onEdit(trans.id, 'debit');
         } else {
-          tableNavigator.onEdit(newId, tableNavigator.focusedField);
+          const newId = onAddSplitProp(id);
+          tableNavigator.onEdit(
+            newId,
+            latestState.current.tableNavigator.focusedField,
+          );
         }
-      }
-    };
-  }, [onSplitProp, splitsExpandedDispatch]);
-
-  const { onAddSplit: onAddSplitProp } = props;
-
-  const onAddSplit = useCallback(
-    id => {
-      const {
-        tableNavigator,
-        newNavigator,
-        newTransactions: newTrans,
-      } = latestState.current;
-
-      if (isTemporaryId(id)) {
-        const { data, diff } = addSplitTransaction(newTrans, id);
-        setNewTransactions(data);
-        newNavigator.onEdit(
-          diff.added[0].id,
-          latestState.current.newNavigator.focusedField,
+      },
+      [onAddSplitProp],
+    );
+
+    const onDistributeRemainder = useCallback(
+      async (id: TransactionEntity['id']) => {
+        const { transactions, newNavigator, tableNavigator, newTransactions } =
+          latestState.current;
+
+        const targetTransactions = isTemporaryId(id)
+          ? newTransactions
+          : transactions;
+        const transaction = targetTransactions.find(t => t.id === id);
+
+        const parentTransaction = transaction?.is_parent
+          ? transaction
+          : targetTransactions.find(t => t.id === transaction?.parent_id);
+
+        const siblingTransactions = targetTransactions.filter(
+          t =>
+            t.parent_id &&
+            t.parent_id ===
+              (transaction?.is_parent
+                ? transaction?.id
+                : transaction?.parent_id),
         );
-      } else {
-        const newId = onAddSplitProp(id);
-        tableNavigator.onEdit(
-          newId,
-          latestState.current.tableNavigator.focusedField,
+
+        const emptyTransactions = siblingTransactions.filter(
+          t => t.amount === 0,
         );
-      }
-    },
-    [onAddSplitProp],
-  );
+        if (!parentTransaction) {
+          console.warn(
```

Comment:
Thinking we should make this an error (`console.error`, not hard crash) as it probably would be a bug, and users might miss a warning in the logs.

Reply:
Sure, I'll change it